lib/matplotlib/tests/test_triangulation.py

```diff
@@ -1115,3 +1115,27 @@ def test_qhull_large_offset():
     triang = mtri.Triangulation(x, y)
     triang_offset = mtri.Triangulation(x + offset, y + offset)
     assert len(triang.triangles) == len(triang_offset.triangles)
+
+
+def test_tricontour_non_finite_z():
+    # github issue 10167.
+    x = [0, 1, 0, 1]
+    y = [0, 0, 1, 1]
+    triang = mtri.Triangulation(x, y)
+    plt.figure()
+
+    with pytest.raises(ValueError) as excinfo:
```

Comment:
To elaborate on the below comment, this can just be
```python
with pytest.raises(ValueError, match='z array cannot contain non-finite values'):
```
and the match line below can be removed

Reply:
@timhoffm and @dstansby Thanks for the info on the match parameter in `pytest.raises`. I've changed the test code to use this improved approach.